modules/Students/student_view_details.php

```diff
@@ -627,14 +627,34 @@
 
                         //Get and display a list of student's teachers
                         echo '<h4>';
-                        echo __("Student's Teachers");
+                        echo sprintf(__('Teachers Of %1$s'), $row['preferredName']);
```

Comment:
This would be a good opportunity for the new translation format, which helps indicate the context of the string replacement.

```suggestion
                        echo __('Teachers Of {student}', ['student' => $row['preferredName']]);
```

Reply:
Thanks for the suggestion. I've implemented this, and committed it.